gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1315 @@
+            elif isinstance(target[0], basemodel.BaseTopicModel):
+                ttda = np.concatenate([model.get_topics() for model in target], axis=0)
+                detected_num_models = len(target)
+
+            # unknown
+            else:
+                raise ValueError("target is of unknown type or a list of unknown types: {}".format(type(target[0])))
+
+            # new models were added, increase num_models
+            # if the user didn't provide a custon numer to use
+            if num_new_models is None:
+                self.num_models += detected_num_models
+            else:
+                self.num_models += num_new_models
+
+        else:  # memory unfriendly ensembles
+            ttda = []
+
+            # 1. ttda array
+            if isinstance(target.dtype.type(), (np.number, float)):
+                raise ValueError(
+                    'ttda arrays cannot be added to ensembles, for which memory_friendly_ttda=False, '
+                    'you can call convert_to_memory_friendly, but it will discard the stored gensim '
+                    'models and only keep the relevant topic term distributions from them.')
+
+            # 2. list of ensembles
+            elif isinstance(target[0], type(self)):
+                for ensemble in target:
+                    self.tms += ensemble.tms
+                ttda = np.concatenate([ensemble.ttda for ensemble in target], axis=0)
+
+            # 3. list of gensim models
+            elif isinstance(target[0], basemodel.BaseTopicModel):
+                self.tms += target.tolist()
+                ttda = np.concatenate([model.get_topics() for model in target], axis=0)
+
+            # unknown
+            else:
+                raise ValueError("target is of unknown type or a list of unknown types: {}".format(type(target[0])))
+
+            # in this case, len(self.tms) should
+            # always match self.num_models
+            if num_new_models is not None and num_new_models + self.num_models != len(self.tms):
+                logger.info(
+                    'num_new_models will be ignored. num_models should match the number of '
+                    'stored models for a memory unfriendly ensemble'
+                )
+            self.num_models = len(self.tms)
+
+        logger.info("ensemble contains {} models and {} topics now".format(self.num_models, len(self.ttda)))
+
+        if self.ttda.shape[1] != ttda.shape[1]:
+            raise ValueError(
+                "target ttda dimensions do not match. Topics must be {} but was {} elements large".format(
+                    self.ttda.shape[-1], ttda.shape[-1],
+                )
+            )
+        self.ttda = np.append(self.ttda, ttda, axis=0)
+
+        # tell recluster that the distance matrix needs to be regenerated
+        self.asymmetric_distance_matrix_outdated = True
+
+    def _teardown(self, pipes, processes, i):
+        """close pipes and terminate processes
```

Comment:
```suggestion
        """Close pipes and terminate processes.
```

Docstrings as full sentences: please capitalize / punctuate.

Reply:
This is changed. 

I did a quick check through and the other doc-strings I saw seemed to conform but I'll keep my eyes open anyways.   @sezanzeb, maybe also take a look through.

Reply:
Tried to search through all of the new functions, and if I didn't miss one they were all in imperative mood and ended with a dot